test/common-test/pipeline/pipeline-test.cpp

```diff
@@ -631,4 +662,90 @@ TEST(ModifyingPipelineTest, TestSequentialAndModifierPipelinesinSequentialPipeli
     ASSERT_EQ(expectedResource, resource);
 }
 
+// Alias-based smoke test for DistancePipeline to ensure the typedef stays wired
+TEST(SequentialPipelineTest, RunDistancePipelineAlias) {
+    DistancePipeline pipeline;
+    ImageToPointsStage stage1;
+    PointsToVec3Stage stage2;
+
+    Image input{};
+    PositionVector result = pipeline.AddStage(stage1).Complete(stage2).Run(input);
+
+    ASSERT_EQ(result.x, 5);
+    ASSERT_EQ(result.y, 6);
+    ASSERT_EQ(result.z, 7);
+}
+
+TEST(SequentialPipelineTest, DoActionImageToVec3N10) {
+    SequentialPipeline<Image, Vec3, 10> pipeline;
+    ImageToPointsStage stage1;
+    PointsToVec3Stage stage2;
+
+    pipeline.AddStage(stage1).Complete(stage2);
+
+    Image input{};
+    Vec3 output{};
+    pipeline.GetResource() = input;
+    pipeline.GetProduct() = &output;
+
+    pipeline.DoAction(); // This directly calls the target function.
+
+    ASSERT_EQ(output.x, 5);
+    ASSERT_EQ(output.y, 6);
+    ASSERT_EQ(output.z, 7);
+}
+
+// Directly exercise FunctionStage<Points, Points>::DoAction
+TEST(FunctionStageTest, DoActionPointsIdentityStage) {
+    IdentityPointsStage stage;
+    Points input = {{DECIMAL(1.0), DECIMAL(2.0)}, {DECIMAL(3.0), DECIMAL(4.0)}};
+    Points product;
+
+    stage.GetResource() = input;
+    stage.GetProduct() = &product;
+
+    stage.DoAction();
+
+    ASSERT_EQ(product.size(), input.size());
+    for (size_t i = 0; i < product.size(); ++i) {
+        EXPECT_EQ(product[i].x, input[i].x + DECIMAL(1.0));
+        EXPECT_EQ(product[i].y, input[i].y + DECIMAL(2.0));
+    }
+}
+
+// Explicit coverage for SequentialPipeline<Image, Vec3, 4>::Run via a member pointer call
+TEST(SequentialPipelineTest, RunImageToVec3N4CoversRun) {
```

Comment:
I'm more confused why this is necessary here? I'm not convinced this is really doing anything for coverage.